src/nvim/api/autocmd.c

```diff
@@ -0,0 +1,359 @@
+#include <stdbool.h>
+
+#include "lauxlib.h"
+#include "nvim/api/autocmd.h"
+#include "nvim/api/private/defs.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/fileio.h"
+#include "nvim/lua/executor.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "api/autocmd.c.generated.h"
+#endif
+
+#define FOREACH_KEY_VALUE(d, k, v, code) \
+  for (size_t i = 0; i < d.size; i++) { \
```

Comment:
we might not want `i` in macro definition due to scope clashes (khash uses `__i` for instance).